erpnext/accounts/report/statement_of_account/statement_of_account.py

```diff
@@ -273,4 +281,40 @@ def get_addresses(company=None, party_type=None, party=None):
 	return {
 		"company": company_addr,
 		"party": party_addr
-	}
\ No newline at end of file
+	}
+
+@frappe.whitelist()
+def notify_party(filters, report, html=None):
+	filters = frappe._dict(json.loads(filters))
+	report = frappe._dict(json.loads(report))
+	attachments = [frappe.attach_print(report.doctype, report.report_name, html=html)]
+	party = frappe.db.get_value(filters.party_type, filters.party, "email_id")
+
+	if not party:
+		frappe.throw(_("Email id is not mentioned in the Customer master for {0}.").format(frappe.bold(filters.party)))
```

Comment:
could be a supplier as well right?

Reply:
use party_type

Reply:
@neilLasrado Statement of Account is only for customer. we may add party it will filter it out based on soa for supplier but can you confirm with megan does SOA is available for supplier too.